### PR TITLE
Persist currentCurrency from state when re-initialising controller

### PR DESCRIPTION
**Pull request by rickycodes**
re: https://github.com/MetaMask/metamask-extension/issues/6710

re: the pinning, when I installed this I got tsc `Version 3.6.4`

which resulted in the following:

```
src/util.ts:113:4 - error TS2322: Type 'any' is not assignable to type 'never'.

113  		normalizedTransaction[key] = NORMALIZERS[key](transaction[key]);
     		~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.
```

builds fine with `3.5.3`

### CODE REVIEW

**Comment by rickycodes** [comment]
I'm going to add a test case for this in a moment :)

File: package.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "tslint": "^5.10.0",
     "tslint-config-prettier": "^1.13.0",
     "typedoc": "^0.15.0",
-    "typescript": "^3.5.3"
+    "typescript": "3.5.3"
```
Comment (danjm) [inline package.json]: Does this change have anything to do with the problem this PR resolves?

File: src/assets/CurrencyRateController.ts [hunk 2]
```diff
@@ -65,7 +69,7 @@ export class CurrencyRateController extends BaseController<CurrencyRateConfig, C
 	constructor(config?: Partial<CurrencyRateConfig>, state?: Partial<CurrencyRateState>) {
 		super(config, state);
 		this.defaultConfig = {
-			currentCurrency: 'usd',
+			currentCurrency: this.getCurrentCurrencyFromState(state),
```
Comment (danjm) [inline src/assets/CurrencyRateController.ts]: So why exactly does this work? Where is `this.defaultState` used?

**Comment by brunobar79** [comment]
@estebanmino didn’t we fix this on the app side while passing the initial state to the engine?

File: package.json (hunk 1 above)
Comment (danjm) [reply package.json]: Oh, I see your comment on this above. So this was just necessary to get the project to build correctly?

Comment (rickycodes) [reply package.json]: this change isn't related, but as I mentioned above I wasn't able to get the project to build without it...

File: src/assets/CurrencyRateController.ts (hunk 2 above)
Comment (danjm) [reply src/assets/CurrencyRateController.ts]: Is it that `this.initialize()` calls `this.update(this.initialState);` in the base controller https://github.com/MetaMask/gaba/blob/f9cbda2780da2f7fec651bd2ca25797c3866b0bd/src/BaseController.ts#L93

which then sets the internal state: https://github.com/MetaMask/gaba/blob/f9cbda2780da2f7fec651bd2ca25797c3866b0bd/src/BaseController.ts#L197

Yeah, I think that is it.

**Comment by danjm** [comment]
Looks good to me but I'll wait for Bruno's comments to be addressed before approving.

**Comment by rickycodes** [comment]
@brunobar79 I removed the version pinning, which should negate the need to update the yarn.lock and npm-shrinkwrap.json

**Pull request merged**